Speed up rebuilding patches and reduce diff

stephan-gh opened the pull request:
I was looking the the `git format-patch` man page a while ago and thought these extra command line parameters might help to reduce the ugly random diff when regenerating patches.

See commit description for details.
Regenerated patches with these changes: https://github.com/Minecrell/Paper/commit/410eaf85a776aaf3476aff12725897fab7ea768e

Eventually, some parts of [`cleanupPatches`](https://github.com/PaperMC/Paper/blob/58bc259a18b6f8f4fa0e30f6aa62db22bf0db2f8/scripts/rebuildPatches.sh#L15-L32) could be removed after this change.

aikar reviewed (changes requested):
No on the zero commit. causes huge problems when doing upstream updates (git uses that data to help apply patches when the line numbers no longer line up), but the other things are nice.

stephan-gh committed:
scripts/rebuildPatches.sh: Reduce diff when regenerating patches

Add --zero-commit --full-index --no-signature to git format-patch
to reduce the diff when doing minor changes in patches.

 --zero-commit: Print 00000000... commit hash in first "From" line
   of the patch. This commit hash is useless because it changes
   every time the patch is applied.

 --full-index: Print full SHA-1 hash on "index aaaa..bbbb" lines.
   This makes the lines much longer, but avoids random changes to
   add/remove a character based on the number of objects that exist
   in the local repository.

 --no-signature: Omit Git version from generated patches.
